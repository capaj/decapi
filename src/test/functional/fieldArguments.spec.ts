import {
  Query,
  SchemaRoot,
  compileSchema,
  ObjectType,
  Arg,
  Field,
} from '~/domains';
import { graphql } from 'graphql';

@ObjectType()
class Hello {
  @Field()
  boolTest(v1: boolean): boolean {
    return v1;
  }
  @Field()
  boolTest2(@Arg({ isNullable: true }) v2: boolean): boolean {
    return v2;
  }
  @Field()
  boolTest3(v3: boolean): boolean {
    return v3;
  }
}

@SchemaRoot()
class FooSchema {
  @Query()
  hello(): Hello {
    return new Hello();
  }
}

const schema = compileSchema({ roots: [FooSchema] });

describe('Query args', () => {
  it('will have correct values even for "false"', async () => {
    const result = await graphql(
      schema,
      `
        {
          hello {
            boolTest(v1: false)
            boolTest2(v2: false)
            boolTest3(v3: true)
          }
        }
      `,
    );

    expect(result.errors).toBeUndefined();
    expect(result.data.hello).toMatchSnapshot();
  });
});
