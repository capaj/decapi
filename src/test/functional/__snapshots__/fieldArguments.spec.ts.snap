// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Query args will have correct values even for "false" 1`] = `
Object {
  "boolTest": false,
  "boolTest2": false,
  "boolTest3": true,
}
`;
